src/server/scripts/EasternKingdoms/BlackrockMountain/BlackwingLair/boss_nefarian.cpp

```diff
@@ -184,18 +153,15 @@ class boss_victor_nefarius : public CreatureScript
         {
             Initialize();
 
-            if (me->GetMapId() == 469)
-            {
-                if (!me->FindNearestCreature(NPC_NEFARIAN, 1000.0f, true))
-                    _Reset();
-
-                me->SetVisible(true);
-                me->SetPhaseMask(1, true);
-                me->SetUInt32Value(UNIT_NPC_FLAGS, 1);
-                me->SetFaction(FACTION_FRIENDLY);
-                me->SetStandState(UNIT_STAND_STATE_SIT_HIGH_CHAIR);
-                me->RemoveAura(SPELL_NEFARIANS_BARRIER);
-            }
+            if (!me->FindNearestCreature(NPC_NEFARIAN, 1000.0f, true))
```

Comment:
thats a bit too much lol

Reply:
it's a big chamber!

Reply:
1000 yards is just the distance from Darnasus to Ashenvale give or take

Reply:
How much (reasonably) is max distance inside the chamber?

Reply:
I measured ingame the distance from the gate you enter to the wall behind Nefarians throne to be ~154 yards, now please review the UBRS script of Nefarius instead :stuck_out_tongue:

Reply:
Its instance script, use instance script instead of grid search ;p

Reply:
sigh, fine, gimme a sec

Reply:
It's just grid calls are expensive and they should be avoided in cases like that, you can pass data via instance script rather easily so yeah, should be done that way :)